lib/socrata.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require 'soda/client'
+
+# This module holds the logic used in lib/tasks/socrata.rake
+module Socrata
+  class << self
+    # Creates a file in the current directory.
+    # Logs and Outputs the name of the file (timestamped).
+    # Returns nothing.
+    def export(file_format, collection)
+      log(:info, 'Exporting dataset...')
+
+      exporter = Decidim::Exporters.find_exporter(file_format || 'CSV')
+      export_data = exporter.new(collection, serializer).export
+      file_name = export_data.filename(collection.first.model_name.plural)
+      File.write(file_name, export_data.read)
+
+      log(:info, "File created: #{file_name}")
+    rescue StandardError => error
+      log(:error, error)
+    end
+
+    # Updates the remote Socrata dataset.
+    # Logs and Outputs the HTTP response.
+    # Returns nothing.
+    #
+    # - client: A SODA::Client instance. Needs to be authenticated.
+    #           https://www.rubydoc.info/github/socrata/soda-ruby/SODA/Client
+    #
+    # - payload: The body of the HTTP POST Request
+    #            Does not accept null values, so we remove them with #compact.
+    #
+    # - identifier: unique name for the dataset we want to access, as in:
```

Comment:
payload and identifier params doesn't exist anymore

Reply:
They do exist at lines 45 and 46.

Reply:
As we've talked, only replace this comments and that's it :+1:

Reply:
Thanks, very helpful.